pywr/parameters/_thresholds.pyx

```diff
@@ -54,6 +57,16 @@ cdef class AbstractThresholdParameter(IndexParameter):
         elif isinstance(predicate, basestring):
             predicate = _predicate_lookup[predicate.upper()]
         self.predicate = predicate
+        self.ratchet = ratchet
+
+    cpdef setup(self):
+        super(AbstractThresholdParameter, self).setup()
+        cdef int ncomb = len(self.model.scenarios.combinations)
+        self._triggered = np.empty(ncomb, dtype=np.int32)
```

Comment:
You can use a boolean dtype here.

Reply:
Apparently `bint` memory views are not possible. Using `uint8` from numpy is the best approach. This is what I've done.